packages/service-survey/src/controllers/statistics.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    Answer,
+    AnswerEnps,
+    AnswerType,
+    enps,
+    Surveys
+} from "@hive/lib-survey";
+import { Request, Response } from "express";
+import { getMongoCollection } from "../mongodb";
+import { HTTPResponse } from "../response";
+
+const isEnpsAnswer = (answer: Answer): answer is AnswerEnps =>
+    answer.type === AnswerType.enum.Enps;
+
+/**
+ * Calculates the eNPS score for a given survey
+ *
+ * @param _req Express response
+ * @param res Express request
+ */
+export const calculateEnps = async (
+    { params: { surveyId } }: Request<{ surveyId: string }>,
+    res: Response
+): Promise<void> => {
+    const allSurveys = Surveys.parse(
+        await getMongoCollection("surveys").find().toArray()
+    );
+
+    // Ensure the survey exists
+    const survey = allSurveys.find(survey => survey._id === surveyId);
+
+    // Get a list of all eNPS scores from eNPS question responses
+    const enpsScores = [];
+    for (const response of survey?.responses ?? []) {
```

Comment:
We could avoid the double loops (O(n^2)) time complexty.
Best to filter answers that are of type `Enps` and then collect the scores

Reply:
What to do if we find no responses in the survey?
What to do if we find no answers of type `Enps`?